XCUITests/PrivateBrowsingTest.swift

```diff
@@ -92,32 +96,40 @@ class PrivateBrowsingTest: BaseTestCase {
         XCTAssertFalse(closePrivateTabsSwitch.isSelected)
 
         //  Open a Private tab
-        navigator.goto(PrivateTabTray)
-        navigator.openURL(urlString: url1)
-        //Wait until the page loads
-        waitUntilPageLoad()
-        navigator.nowAt(PrivateBrowserTab)
-        navigator.goto(PrivateTabTray)
+        navigator.goto(TabTray)
+        navigator.performAction(Action.TogglePrivateMode )
+        XCTAssertTrue(navigator.userState.isPrivate, "User is private")
+        navigator.openURL(url1, waitForLoading: true)
 
         // Go back to regular browser
         navigator.goto(TabTray)
+        navigator.performAction(Action.TogglePrivateMode )
+        XCTAssertFalse(navigator.userState.isPrivate, "User is private")
 
         // Go back to private browsing and check that the tab has not been closed
-        navigator.goto(PrivateTabTray)
+        navigator.goto(TabTray)
+        navigator.performAction(Action.TogglePrivateMode )
+        XCTAssertTrue(navigator.userState.isPrivate, "User is private")
+
         waitforExistence(app.collectionViews.cells[url1Label])
         let numPrivTabs = app.collectionViews.cells.count
         XCTAssertEqual(numPrivTabs, 1, "The number of tabs is not correct, the private tab should not have been closed")
 
-        app.collectionViews.cells[url1Label].tap()
-        navigator.nowAt(PrivateBrowserTab)
         // Now the enable the Close Private Tabs when closing the Private Browsing Button
+        app.collectionViews.cells[url1Label].tap()
+        navigator.nowAt(BrowserTab)
```

Comment:
Consider adding an action. (essentially, `nowAt` is an escape hatch when you can't use actions or screen states).